packages/ckeditor5-dev-changelog/src/utils/generatechangelog.ts

```diff
@@ -125,11 +135,20 @@ const main: GenerateChangelogEntryPoint<GenerateChangelogConfig> = async options
 		return newChangelog as any;
 	}
 
-	await removeChangelogEntryFiles( entryPaths );
+	let pathsToCommit = entryPaths;
+
+	// Handle changelog entry files based on release type.
+	if ( releaseType === 'latest' ) {
+		await removeChangelogEntryFiles( entryPaths );
+	} else {
+		const targetChannel = detectReleaseChannel( newVersion );
+		pathsToCommit = await moveChangelogEntryFiles( entryPaths, targetChannel );
```

Comment:
To enable tracking all changes to files from the Git perspective, we should amend the `pathsToCommit` array instead of overriding it.

`pathsToCommit` must include the source files and the destination files.

While the results still operate on the same data, from the tool perspective, we have two collections:

1. Source files that have been moved to another directory,
2. The destination files required for incoming changelog generation.

It's Git's responsibility to commit files,

https://github.com/ckeditor/ckeditor5-dev/blob/79ed963ec28b28b17a08f01072d31449958ca52e/packages/ckeditor5-dev-utils/src/tools/commit.ts#L29-L35